Wikipedia/Code/PreviewAndSaveViewController.m

```diff
@@ -197,6 +205,15 @@ - (void)goForward {
     }
 }
 
+- (void)goToArticleFor:(NSString *)href {
```

Comment:
Instead of adding this logic here, would you mind adding an additional method to the `PreviewAndSaveViewControllerDelegate`?

```
- (void)previewViewControllerDidCancel:(PreviewAndSaveViewController *)previewViewController;

```

This way the delegate can handle any navigation. You also won't need the data stores in this class anymore or need to manually construct the URL.

Thanks and let me know if this makes sense